spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWriterV2.java

```diff
@@ -76,18 +76,17 @@ public void testMergeSchemaFailsWithoutWriterOption() throws Exception {
 
     // this has a different error message than the case without accept-any-schema because it uses
     // Iceberg checks
-    AssertHelpers.assertThrows(
-        "Should fail when merge-schema is not enabled on the writer",
-        IllegalArgumentException.class,
-        "Field new_col not found in source schema",
-        () -> {
-          try {
-            threeColDF.writeTo(tableName).append();
-          } catch (NoSuchTableException e) {
-            // needed because append has checked exceptions
-            throw new RuntimeException(e);
-          }
-        });
+    Assertions.assertThatThrownBy(
+            () -> {
+              try {
```

Comment:
do we need this try-catch?

Reply:
Thanks comments, it can be remove, same as below